staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/controller/controller.go

```diff
@@ -0,0 +1,274 @@
+	defer d.queue.Add(workqueueKey)
+
+	// return if file has not been removed or renamed.
+	if event.Op&(fsnotify.Remove|fsnotify.Rename) == 0 {
+		return nil
+	}
+
+	if err := watcher.Remove(d.filePath); err != nil {
+		klog.V(2).InfoS("Failed to remove file watch, it may have been deleted", "file", d.filePath, "err", err)
+	}
+	if err := watcher.Add(d.filePath); err != nil {
+		return fmt.Errorf("error adding watch for file %s: %v", d.filePath, err)
+	}
+
+	return nil
+}
+
+// runWorker to process file content
+func (d *DynamicKMSEncryptionConfigContent) runWorker() {
+	for d.processNextWorkItem() {
+	}
+}
+
+// processNextWorkItem processes file content when there is a message in the queue.
+func (d *DynamicKMSEncryptionConfigContent) processNextWorkItem() bool {
+	// key here is dummy item in the queue to trigger file content processing.
+	key, quit := d.queue.Get()
+	if quit {
+		return false
+	}
+	defer d.queue.Done(key)
+
+	var (
+		updatedEffectiveConfig bool
+		err                    error
+	)
+
+	// get context to close the new transformers.
+	ctx, closeTransformers := wait.ContextForChannel(d.stopCh)
+
+	defer func() {
+		if updatedEffectiveConfig {
+			// TODO: increment success metric
+		} else {
+			// avoid leaking if we're not using the newly constructed transformers (due to an error or them not being changed)
+			closeTransformers()
+		}
+		if err != nil {
+			// TODO: increment failure metric
+			utilruntime.HandleError(fmt.Errorf("error processing encryption config file %s: %v", d.filePath, err))
+			// add dummy item back to the queue to trigger file content processing.
+			d.queue.AddRateLimited(key)
+		}
+	}()
+
+	transformerOverrides, kmsPluginHealthzChecks, processedConfigHash, kmsMaxTimeout, configChanged, err := d.processEncryptionConfig(ctx)
+	if err != nil {
+		return true
+	}
+	if !configChanged {
+		return true
+	}
+
+	// get healthz checks for all new KMS plugins.
+	if err = d.validateNewTransformersHealth(kmsPluginHealthzChecks); err != nil {
+		return true
+	}
+
+	// update transformers.
+	d.dynamicTransformers.Set(transformerOverrides, closeTransformers, kmsPluginHealthzChecks, kmsMaxTimeout)
+
+	// update local copy of recent config content once update is successful.
+	d.lastLoadedEncryptionConfigHash = processedConfigHash
+	klog.V(2).InfoS("Loaded new kms encryption config content", "name", d.name)
+
+	updatedEffectiveConfig = true
+	return true
+}
+
+// loadEncryptionConfig processes the next set of content from the file.
+func (d *DynamicKMSEncryptionConfigContent) processEncryptionConfig(ctx context.Context) (
+	transformerOverrides map[schema.GroupResource]value.Transformer,
+	kmsPluginHealthzChecks []healthz.HealthChecker,
+	processedConfigHash string,
+	kmsMaxTimeout time.Duration,
+	configChanged bool,
+	err error,
+) {
+	encryptionConfig, err := d.readEncryptionFileContent()
+	if err != nil {
+		return nil, nil, "", 0, configChanged, err
```

Comment:
isn't default false?

Reply:
Setting a non-zero return value makes me think I need to figure out what that value is on return instead of ignoring it and just looking at the error.  Just like we explicitly pass in `true` for reload, this should explicitly return `false`.